src/cow-react/modules/limitOrders/updaters/QuoteUpdater/hooks/useQuoteRequestParams.ts

```diff
@@ -2,30 +2,50 @@ import { useLimitOrdersTradeState } from '@cow/modules/limitOrders/hooks/useLimi
 import { useWeb3React } from '@web3-react/core'
 import { LegacyFeeQuoteParams as FeeQuoteParams } from '@cow/api/gnosisProtocol/legacy/types'
 import { OrderKind } from '@cowprotocol/contracts'
-import { Token } from '@uniswap/sdk-core'
-import { parseUnits } from '@ethersproject/units'
+import { parseUnits } from 'ethers/lib/utils'
 import { useMemo } from 'react'
+import { useTypedValue } from '@cow/modules/limitOrders/hooks/useTypedValue'
+import { getAddress } from '@cow/modules/limitOrders/utils/getAddress'
+import { calculateValidTo } from '@cow/utils/time'
+import { useUserTransactionTTL } from '@src/state/user/hooks'
 
 export function useQuoteRequestParams(): FeeQuoteParams | null {
-  const { inputCurrency, outputCurrency, inputCurrencyAmount, recipient, deadline } = useLimitOrdersTradeState()
+  const { inputCurrency, outputCurrency, recipient, orderKind } = useLimitOrdersTradeState()
   const { chainId, account } = useWeb3React()
+  const { exactTypedValue } = useTypedValue()
+  const [deadline] = useUserTransactionTTL()
```

Comment:
`useUserTransactionTTL()` - that hook provides data only for Swap. For limit-orders we should use `useLimitOrdersTradeState()`